src/mds/Locker.cc

```diff
@@ -5503,6 +5512,7 @@ void Locker::local_wrlock_finish(const MutationImpl::lock_iterator& it, Mutation
   mut->locks.erase(it);
   if (lock->get_num_wrlocks() == 0) {
     lock->finish_waiters(SimpleLock::WAIT_STABLE |
```

Comment:
`STABLE` and `XLOCK` are aliases for the same bit, there's no need to specify both.

Furthermore, we should not wake up all of these waiters, only those waiting for the `XLOCK`.

Also, in the `local_wrlock_start`, the `else` clause of `if (can_wrlock())`, we should only add a waiter for `WAIT_WR`. I can't comment on that line because it's not part of the patch

Reply:
Thanks for reminding me about WAIT_STABLE. I've updated the commit in the next push.